pkg/oci/skopeo.go

```diff
@@ -33,10 +35,35 @@ func LoadPodman(dir, imageName string) error {
 
 // Push image to registry
 func Push(dir, imageName string) error {
-	cmd := exec.Command("nix", "run", "nixpkgs#skopeo", "--", "copy", "--insecure-policy", "dir:"+dir, "docker://"+imageName)
+
+	currentuser, err := user.Current()
+	if err != nil {
+		return err
+	}
+	authFile := currentuser.HomeDir + "/.skopeo/config.json"
+	cmd := exec.Command("nix", "run", "nixpkgs#skopeo", "--", "copy", "--authfile", authFile, "--insecure-policy", "dir:"+dir, "docker://"+imageName)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
-	err := cmd.Run()
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+func Auth() error {
+	currentuser, err := user.Current()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Logging in to registry")
+	authFile := currentuser.HomeDir + "/.skopeo/config.json"
+	cmd := exec.Command("nix", "run", "nixpkgs#skopeo", "--", "login", "--authfile", authFile, "docker.io")
```

Comment:
can we use the skopeo as a library for login?

Reply:
yep we can